shared/src/commonMain/kotlin/com/automattic/myapplication/shared/internal/database/Database.kt

```diff
@@ -0,0 +1,39 @@
+package com.automattic.myapplication.shared.internal.database
+
+import com.automattic.myapplication.shared.Event
+import com.automattic.myapplication.shared.EventsDatabase
+import com.automattic.myapplication.shared.TracksEvent
+import com.squareup.sqldelight.runtime.coroutines.asFlow
+import com.squareup.sqldelight.runtime.coroutines.mapToList
+import kotlinx.coroutines.flow.Flow
+import kotlin.coroutines.CoroutineContext
+
+internal class Database(
+    driverFactory: DriverFactory,
+    private val coroutineContext: CoroutineContext,
+) {
+
+    private val database = EventsDatabase(driver = driverFactory.createDriver())
+    private val dbQuery = database.eventQueries
+
+    internal fun observeOldest(): Flow<List<Event>> =
+        dbQuery.selectOldest().asFlow().mapToList(coroutineContext)
+
+    internal fun insertEvent(tracksEvent: TracksEvent) {
+        dbQuery.insertEvent(
+            event_name = tracksEvent.name,
+            user = "",
+            user_agent = null,
+            timestamp = null,
+            retry_count = null,
+            user_type = null,
+            user_props = null,
+            device_info = null,
+            custom_props = null
```

Comment:
this structure is taken from [current implementation of Tracks](https://github.com/Automattic/Automattic-Tracks-Android/blob/develop/AutomatticTracks/src/main/java/com/automattic/android/tracks/datasets/EventTable.java#L22-L32), I wasn't sure if it's all needed, probably we'll verify that later.